### PR TITLE
Add cancellation flow and RTL based testing

### PR DESCRIPTION
**Pull request by Tankenstein**
Firstly, this PR adds support for cancelling applications. For this, we reuse the signing functionality built in redux and wrap it within hooks for easy reuse.

Secondly, it adds a setup to be able to start testing larger parts of the app with react testing library, where we mock only the backend. Tests for the cancellation flow are written using this setup.

### CODE REVIEW

File: src/entries/components/common/api.ts [hunk 1]
```diff
@@ -295,88 +289,15 @@ export function getPendingApplications(token: string): Promise<Application[]> {
...
-  pillar: number;
-  managementFeeRate: number;
-  ongoingChargesFigure: number;
-  fundManager: FundManager;
-}
-
-export interface FundManager {
-  id: number;
-  name: string;
+export function createApplicationCancellation(
```
Comment (Tankenstein) [inline src/entries/components/common/api.ts:292]: I moved the api models out to an `apiModels` file to not make this file even larger.

File: src/entries/components/config/config.js [hunk 2]
```diff
@@ -18,6 +18,14 @@ const DEFAULT_CONF_PRODUCTION = {
   language: 'et',
 };
 
+const DEFAULT_CONF_TEST = {
```
Comment (Tankenstein) [inline src/entries/components/config/config.js:21]: Needed this to run larger parts of the app in RTL, which consume this configuration.

File: src/entries/components/exchange/actions.js [hunk 3]
```diff
@@ -267,6 +267,13 @@ function signIdCardSignatureHashWithCertificateForMandateId(hash, certificate, m
       });
 }
 
+function saveOrRetrieveExistingMandate(mandate, token) {
```
Comment (Tankenstein) [inline src/entries/components/exchange/actions.js:270]: This is how we have managed to reuse the signing code for a mandate that already exists

File: src/entries/components/exchange/actions.spec.js [hunk 4]
```diff
@@ -221,12 +221,15 @@ describe('Exchange actions', () => {
...
+      expect(mockApi.saveMandateWithToken).toHaveBeenCalledWith({}, 'token');
       return Promise.resolve(controlCode);
     });
+    mockApi.getMobileIdSignatureStatusForMandateIdWithToken = jest.fn(() =>
+      Promise.resolve({ statusCode: 'OUTSTANDING_TRANSACTION' }),
+    );
     const signMandate = createBoundAction(actions.signMandateWithMobileId);
     expect(dispatch).not.toHaveBeenCalled();
-    await signMandate(mandate);
+    await signMandate({});
```
Comment (Tankenstein) [inline src/entries/components/exchange/actions.spec.js:232]: Had to change these tests as they were giving it an existing mandate - behaviour they would not actually have in current production code.

File: src/entries/components/exchange/actions.spec.js [hunk 5]
```diff
@@ -438,10 +437,44 @@ describe('Exchange actions', () => {
...
+    state.login.loginMethod = 'smartId';
+    const signMandate = createBoundAction(actions.signMandate);
+    mockApi.saveMandateWithToken = jest.fn(() => Promise.resolve(mandate));
+    mockApi.getSmartIdSignatureStatusForMandateIdWithToken = jest.fn(() =>
+      Promise.reject(new Error('oh no')),
+    );
+    await signMandate(mandate);
+    expect(mockApi.saveMandateWithToken).not.toHaveBeenCalled();
+  });
+
```
Comment (Tankenstein) [inline src/entries/components/exchange/actions.spec.js:458]: Oops, forgot to comment this back in, fixing.

File: package.json [hunk 6]
```diff
@@ -106,8 +106,8 @@
       "pre-commit": "lint-staged"
     }
   },
-  "proxy": "http://localhost:9000",
-  "//": "https://onboarding-service.tuleva.ee",
+  "//": "http://localhost:9000",
+  "proxy": "https://onboarding-service.tuleva.ee",
```
Comment (ErkoRisthein) [inline package.json:110]: Revert

File: src/entries/components/exchange/actions.spec.js [hunk 7]
```diff
@@ -438,10 +437,50 @@ describe('Exchange actions', () => {
...
+    const mandate = { id: 'id' };
+    state.login.loginMethod = 'idCard';
+    const signMandate = createBoundAction(actions.signMandate);
+    mockApi.saveMandateWithToken = jest.fn(() => Promise.resolve(mandate));
+
+    const certificate = { hex: 'certificate' };
+    global.hwcrypto = mockHwcrypto;
+    mockHwcrypto.getCertificate = jest.fn(() => Promise.resolve(certificate));
+    mockApi.getIdCardSignatureHashForMandateIdWithCertificateHexAndToken = jest.fn(() =>
+      Promise.reject(new Error('oh no')),
```
Comment (ErkoRisthein) [inline src/entries/components/exchange/actions.spec.js:481]: `new Error('stop polling'))`

File: src/entries/components/flows/cancellation/ConfirmCancellation.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    cancelApplication(applicationId);
+  }
+
+  function previewCancellation() {
+    downloadPreview(applicationId);
+  }
+
+  useEffect(() => {
+    if (signedMandateId) {
+      push(
```
Comment (ErkoRisthein) [inline src/entries/components/flows/cancellation/ConfirmCancellation.tsx:33]: Replace with redirect component, if possible

File: src/entries/components/flows/common/SuccessNotice.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import styles from './Success.module.scss';
+import successImage from './success.svg';
+
+export const SuccessNotice: React.FunctionComponent<{ children?: React.ReactNode }> = ({
```
Comment (ErkoRisthein) [inline src/entries/components/flows/common/SuccessNotice.tsx:5]: Maybe extract to `common/success` package

**Review by ErkoRisthein — APPROVED** [review]
🍺

**Pull request merged**